lib/dom.js

```diff
@@ -1060,6 +1068,167 @@ Node.prototype = {
 	 * @type {string | null}
 	 */
 	localName: null,
+	/**
+	 * The URI is currently always `about:blank` since that's what happens when you create a
+	 * document from scratch.
+	 * This could be improved by making it possible to add a baseURI to the document when creating
+	 * one so it's childNodes just use it's URI instead.
+	 *
+	 * @type {string}
+	 */
+	baseURI: 'about:blank',
+	/**
+	 * Is true if this node is part of a document.
+	 *
+	 * @type {boolean}
+	 */
+	get isConnected() {
+		var rootNode = this.getRootNode();
+		if (rootNode && rootNode.nodeType === rootNode.DOCUMENT_NODE) {
+			return true;
+		}
+
+		return false;
+	},
+	/**
+	 * Checks whether or not `other` is an inclusive descendant of this node.
+	 *
+	 * @param {Node} [other]
+	 * The node to check.
+	 * @returns {boolean}
+	 * True if `other` is an inclusive descendant of this node; false otherwise.
+	 * @see https://dom.spec.whatwg.org/#dom-node-contains
+	 */
+	contains: function (other) {
+		if (!other) {
+			return false;
+		}
+
+		do {
+			if (this === other) {
+				return true;
+			}
+		} while ((other = other.parentNode));
+
+		return false;
+	},
+	/**
+	 * @typedef GetRootNodeOptions
+	 * @property {boolean} [composed=false]
+	 */
+	/**
+	 * Searches for the root node of this node.
+	 *
+	 * Difference to the specs:
+	 * options.compose doesn't change this behaviour.
+	 *
+	 * @param {GetRootNodeOptions} [options]
+	 * @returns {Node}
+	 * Root node.
+	 * @see https://dom.spec.whatwg.org/#dom-node-getrootnode
+	 */
+	getRootNode: function (options) {
+		if (options && options.composed) {
+			// should return shadow root here (not supported)
+		}
+
+		var parent = this;
+		do {
+			if (!parent.parentNode) {
+				return parent;
+			}
+		} while ((parent = parent.parentNode));
+	},
+	/**
+	 * Checks whether or not the given node is equal to this node.
+	 *
+	 * @param {Node} [otherNode]
+	 * @see https://dom.spec.whatwg.org/#concept-node-equals
+	 */
+	isEqualNode: function (otherNode) {
+		if (!otherNode) {
+			return false;
+		}
+
+		if (this.nodeType !== otherNode.nodeType) {
+			return false;
+		}
+
+		if (
+			(this.childNodes && otherNode.childNodes && this.childNodes.length !== otherNode.childNodes.length) ||
+			(!this.childNodes && otherNode.childNodes) ||
+			(this.childNodes && !otherNode.childNodes)
```

Comment:
All `Node`s have a `childNodes` attribute, right? How can it not be set?

I think it is because we do not really have classes and inheritance so the property is not set from all the "constructors"...

We could fix it in this PR, by setting `childNodes` in all contructors, with the benefit of simpifying this check to
`this.childNodes.length !== otherNode.childNodes.length`,
or we keep it for another PR and will file an issue.

What do you think?

Reply:
FYI: I added `childNodes = new NodeList()` for every `Document.create*() => Node` method, but kept the condition for now, we will see if it's possible to produce test cases for all the missing cases.